### PR TITLE
chore(RUN): Upgrade wasmtime to 24

### PR DESCRIPTION
**Pull request by adambratschikaye**
Release notes here: https://github.com/bytecodealliance/wasmtime/releases/tag/v24.0.0

I don't see any changes relevant to the IC.

### CODE REVIEW

**Comment by adambratschikaye** [comment]
Let's wait until September 5th or so to merge this to see if any patch releases come out.

**Review by cgundy — DISMISSED** [review]

File: Cargo.lock
Comment (adambratschikaye) [reply Cargo.lock:21154]: In the past we've kept the version of these crates which we're directly using in sync with the version used by wasmtime, but we bump them in separate follow-up PRs because they can require messy changes (as new Wasm features are added) and we don't want them to block the regular wasmtime upgrades.

So how about doing these separately?

**Review by nmattia — APPROVED** [review]

**Review by cgundy — APPROVED** [review]

File: Cargo.Bazel.Fuzzing.json.lock
Comment (adambratschikaye) [reply Cargo.Bazel.Fuzzing.json.lock:24392]: I'm not sure how much luck we're going to have syncing up the different versions of `gimli`. It seems that one comes from `wasmtime`, one comes from `walrus`, and one comes from the dependency chain `gimli -> addr2line -> backtrace` which is used by both `tokio` and `pprof`.

**Pull request merged**
